File: 0461-hamming-distance/0461-hamming-distance.py
class Solution:
    def getBinary(self, n :int) -> list[int]:
        binary = [0 for i in range(32)]
        i = 0
        while n > 0:
            binary[i] = n%2
            n = n//2
            i += 1
        return binary

    def hammingDistance(self, x: int, y: int) -> int:
        answer = 0
        for i,j in zip(self.getBinary(x), self.getBinary(y)):
            if i != j:
                answer += 1
        return answer

File: 0461-hamming-distance/0461-hamming-distance.go
import "fmt"
func hammingDistance(x int, y int) int {
    answer := 0
    x_b := getBinary(x)
    y_b := getBinary(y)
    for i:= 0;i < 32; i++ {
        if x_b[i] != y_b[i] {
            answer++
        }
    }
    return answer
}

func getBinary(n int) []int {
    binary := make([]int, 32, 32)
    i := 0
    for {
        binary[i] = n%2
        i++
        n = int(n/2)
        if n == 0 {
            break
        }
    }
    return binary
}
